src/Microsoft.Owin.Security.Authorization/IAuthorizationDependenciesProvider.cs

```diff
@@ -0,0 +1,11 @@
+using System;
+
+namespace Microsoft.Owin.Security.Authorization
+{
+    public interface IAuthorizationDependenciesProvider
+    {
+        Func<AuthorizationOptions, IOwinContext, AuthorizationDependencies> OnCreate { get; }
+
+        Action<AuthorizationOptions, IOwinContext, AuthorizationDependencies> OnDispose { get; }
```

Comment:
If a middleware must handle an `IDisposable` object directly, it should generally take a factory which creates and returns the necessary `IDisposable` object.  That way, ownership and lifetime of the `IDisposable` is entirely contained within the middleware itself, and the many, *many* obscure edge cases involving transfer of `IDisposable` ownership are avoided.  (For example, passing an `IDisposable` as a function argument technically allows for the function to dispose it even while the caller maintains a strong reference to it from outside, so the caller can end up referencing an object in a disposed state, double disposing, etc.).